File: app/src/main/java/com/adiaz/deportesmadrid/db/daos/GroupsDAO.java
package com.adiaz.deportesmadrid.db.daos;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

import com.adiaz.deportesmadrid.db.entities.Group;
import com.adiaz.deportesmadrid.retrofit.groupslist.GroupRetrofitEntity;
import com.adiaz.deportesmadrid.utils.Constants;

import static com.adiaz.deportesmadrid.db.DbContract.GroupEntry;

import java.util.ArrayList;
import java.util.List;

public class GroupsDAO {


    public static void insertCompetitions(Context context, List<GroupRetrofitEntity> competitionsList) {
        List<ContentValues> contentValues = new ArrayList<>();
        for (GroupRetrofitEntity GroupRetrofitEntity : competitionsList) {
            if (GroupRetrofitEntity.getDistrito().equalsIgnoreCase(Constants.DGD)) {
                GroupRetrofitEntity.setDistrito(Constants.DISTRITO_UNICO);
            }
            ContentValues cv = GroupEntry.retrofitEntityToContentValue(GroupRetrofitEntity);
            contentValues.add(cv);
        }
        ContentValues[] array = contentValues.toArray(new ContentValues[contentValues.size()]);

        //before insert it is necessary to delete all.
        context.getContentResolver().delete(GroupEntry.CONTENT_URI, null, null);
        context.getContentResolver().bulkInsert(GroupEntry.CONTENT_URI, array);
    }

    public static List<Group> queryAllCompetitions(Context context) {
        List<Group> groups = new ArrayList<>();
        Cursor cursor = null;
        try {
            cursor = context.getContentResolver().query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, null, null, null);
            while (cursor!=null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor));
            }
        } finally {
            if (cursor!=null) {
                cursor.close();
            }
        }
        return groups;
    }

    public static Group queryCompetitionsById(Context context, String idCompetion) {
        String selection = GroupEntry.COLUMN_ID + "=?";
        String[] selectionArgs = new String[]{idCompetion};
        Cursor cursor = null;
        try {
            cursor = context.getContentResolver().query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null);
            if (cursor!=null && cursor.moveToNext()) {
                return GroupEntry.cursorToEntity(cursor);
            }
        } finally {
            if (cursor!=null) {
                cursor.close();
            }
        }
        return null;
    }

    public static List<Group> queryCompetitionsBySport(Context context, String sportName) {
        List<Group> groups = new ArrayList<>();
        String selection = GroupEntry.COLUMN_DEPORTE + "=?";
        String[] selectionArgs = new String[]{sportName};
        Cursor cursor = null;
        try {
            cursor = context.getContentResolver().query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null);
            while (cursor!=null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor));
            }
        } finally {
            if (cursor!=null) {
                cursor.close();
            }
        }
        return groups;
    }

    public static List<Group> queryCompetitionsBySportAndDistrict(Context context, String sportName, String district) {
        List<Group> groups = new ArrayList<>();
        String selection = GroupEntry.COLUMN_DEPORTE + "=?";
        selection += "and " + GroupEntry.COLUMN_DISTRITO + "=?";
        String[] selectionArgs = new String[]{sportName, district};
        Cursor cursor = null;
        try {
            cursor = context.getContentResolver().query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null);
            while (cursor!=null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor));
            }
        } finally {
            if (cursor != null) {
                cursor.close();
            }
        }
        return groups;
    }

    public static List<Group> queryCompetitionsBySportAndDistrictAndCategory(Context context, String sportName, String district, String category) {
        List<Group> groups = new ArrayList<>();
        String selection = GroupEntry.COLUMN_DEPORTE + "=?";
        selection += "and " + GroupEntry.COLUMN_DISTRITO + "=?";
        selection += "and " + GroupEntry.COLUMN_CATEGORIA + "=?";
        String[] selectionArgs = new String[]{sportName, district, category};
        Cursor cursor = null;
        try {
            cursor = context.getContentResolver().query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null);
            while (cursor!=null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor));
            }
        } finally {
            if (cursor!=null) {
                cursor.close();
            }
        }
        return groups;
    }
}

File: app/src/main/java/com/adiaz/ligasmadrid/db/daos/GroupsDAO.kt
package com.adiaz.ligasmadrid.db.daos

import android.content.ContentValues
import android.content.Context
import android.database.Cursor

import com.adiaz.ligasmadrid.db.entities.Group
import com.adiaz.ligasmadrid.retrofit.groupslist.GroupRetrofitEntity
import com.adiaz.ligasmadrid.utils.Constants

import com.adiaz.ligasmadrid.db.DbContract.GroupEntry

import java.util.ArrayList

object GroupsDAO {


    fun insertCompetitions(context: Context, competitionsList: List<GroupRetrofitEntity>) {
        val contentValues = ArrayList<ContentValues>()
        for (GroupRetrofitEntity in competitionsList) {
            if (GroupRetrofitEntity.distrito.equals(Constants.DGD, ignoreCase = true)) {
                GroupRetrofitEntity.distrito = Constants.DISTRITO_UNICO
            }
            val cv = GroupEntry.retrofitEntityToContentValue(GroupRetrofitEntity)
            contentValues.add(cv)
        }
        val array = contentValues.toTypedArray()

        //before insert it is necessary to delete all.
        context.contentResolver.delete(GroupEntry.CONTENT_URI, null, null)
        context.contentResolver.bulkInsert(GroupEntry.CONTENT_URI, array)
    }

    fun queryAllCompetitions(context: Context): List<Group> {
        val groups = ArrayList<Group>()
        var cursor: Cursor? = null
        try {
            cursor = context.contentResolver.query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, null, null, null)
            while (cursor != null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor))
            }
        } finally {
            cursor?.close()
        }
        return groups
    }

    fun queryCompetitionsById(context: Context, idCompetion: String): Group? {
        val selection = GroupEntry.COLUMN_ID + "=?"
        val selectionArgs = arrayOf(idCompetion)
        var cursor: Cursor? = null
        try {
            cursor = context.contentResolver.query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null)
            if (cursor != null && cursor.moveToNext()) {
                return GroupEntry.cursorToEntity(cursor)
            }
        } finally {
            cursor?.close()
        }
        return null
    }

    fun queryCompetitionsBySport(context: Context, sportName: String): List<Group> {
        val groups = ArrayList<Group>()
        val selection = GroupEntry.COLUMN_DEPORTE + "=?"
        val selectionArgs = arrayOf(sportName)
        var cursor: Cursor? = null
        try {
            cursor = context.contentResolver.query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null)
            while (cursor != null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor))
            }
        } finally {
            cursor?.close()
        }
        return groups
    }

    fun queryCompetitionsBySportAndDistrict(context: Context, sportName: String, district: String): List<Group> {
        val groups = ArrayList<Group>()
        var selection = GroupEntry.COLUMN_DEPORTE + "=?"
        selection += "and " + GroupEntry.COLUMN_DISTRITO + "=?"
        val selectionArgs = arrayOf(sportName, district)
        var cursor: Cursor? = null
        try {
            cursor = context.contentResolver.query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null)
            while (cursor != null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor))
            }
        } finally {
            cursor?.close()
        }
        return groups
    }

    fun queryCompetitionsBySportAndDistrictAndCategory(context: Context, sportName: String, district: String, category: String): List<Group> {
        val groups = ArrayList<Group>()
        var selection = GroupEntry.COLUMN_DEPORTE + "=?"
        selection += "and " + GroupEntry.COLUMN_DISTRITO + "=?"
        selection += "and " + GroupEntry.COLUMN_CATEGORIA + "=?"
        val selectionArgs = arrayOf(sportName, district, category)
        var cursor: Cursor? = null
        try {
            cursor = context.contentResolver.query(GroupEntry.CONTENT_URI, GroupEntry.PROJECTION, selection, selectionArgs, null)
            while (cursor != null && cursor.moveToNext()) {
                groups.add(GroupEntry.cursorToEntity(cursor))
            }
        } finally {
            cursor?.close()
        }
        return groups
    }
}
